### PR TITLE
Add jest completion

### PR DESCRIPTION
**Pull request by aoyama-val**
- [x] This compdef is not already available in zsh.
- [x] This compdef is not already available in their original project.
- [x] I am the original author, or I have authorization to submit this work.
- [x] This is a finished work.
- [x] It has a header containing authors, status and origin of the script.
- [x] It has a license header or I accept that it will be licensed under the terms of the Zsh license.

### CODE REVIEW

File: src/_jest [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# ------------------------------------------------------------------------------
+# Description
+# -----------
+#
+#  Completion script for Node.js 29.7.0 (https://jestjs.io/)
```
Comment (syohex) [inline src/_jest:31]:
```suggestion
#  Completion script for Jest 29.7.0 (https://jestjs.io/)
```

File: src/_jest [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  --changedSince'[Runs tests related to the changes since the provided branch. If the current branch has diverged from the given branch, then only changes made locally will be tested. Behaves similarly to `--onlyChanged`. (string)]:string' \
+  --ci'[Whether to run Jest in continuous integration (CI) mode. This option is on by default in most popular CI environments. It will prevent snapshots from being written unless explicitly requested. (boolean)]' \
+  --clearCache'[Clears the configured Jest cache directory and then exits. Default directory can be found by calling jest --showConfig (boolean)]' \
+  --clearMocks'[Automatically clear mock calls, instances, contexts and results before every test. Equivalent to calling jest.clearAllMocks() before each test. (boolean)]' \
+  --collectCoverage'[Alias for --coverage. (boolean)]' \
+  --collectCoverageFrom'[A glob pattern relative to <rootDir> matching the files that coverage info needs to be collected from. (string)]:string' \
+  --color'[Forces test results output color highlighting (even if stdout is not a TTY). Set to false if you would like to have no colors. (boolean)]' \
+  --colors'[Alias for `--color`. (boolean)]' \
```
Comment (syohex) [inline src/_jest:60]:
```suggestion
 '(--color --colors)'{--color,--colors}'[Forces test results output color highlighting (even if stdout is not a TTY). Set to false if you would like to have no colors. (boolean)]' \
```

We should not specify both `--color` and `--colors` so they should be merged. Or remove the alias version

File: src/_jest [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  --cacheDirectory'[The directory where Jest should store its cached dependency information. (string)]:directories:_dirs' \
+  --changedFilesWithAncestor'[Runs tests related to the current changes and the changes made in the last commit. Behaves similarly to `--onlyChanged`. (boolean)]' \
+  --changedSince'[Runs tests related to the changes since the provided branch. If the current branch has diverged from the given branch, then only changes made locally will be tested. Behaves similarly to `--onlyChanged`. (string)]:string' \
+  --ci'[Whether to run Jest in continuous integration (CI) mode. This option is on by default in most popular CI environments. It will prevent snapshots from being written unless explicitly requested. (boolean)]' \
+  --clearCache'[Clears the configured Jest cache directory and then exits. Default directory can be found by calling jest --showConfig (boolean)]' \
+  --clearMocks'[Automatically clear mock calls, instances, contexts and results before every test. Equivalent to calling jest.clearAllMocks() before each test. (boolean)]' \
+  --collectCoverage'[Alias for --coverage. (boolean)]' \
```
Comment (syohex) [inline src/_jest:57]: Please see the comment about `--color` and `--colors`

**Comment by syohex** [comment]
image

Users cannot see a description as above if it is too long. So I think it should be one sentence at most except special cases.In most cases, it is enough to use the first sentence as the flag description.  And I think type information is redundant

**Comment by aoyama-val** [comment]
Thank you for your review. Fixed them. I made the descriptions shorter, but the longest one is still 146 characters long. I don't think that's a real harml, rather benefit for users who have large monitors. What do you think?

**Comment by syohex** [comment]
I think it's ok. thanks

**Pull request merged**